airflow/utils/helpers.py

```diff
@@ -116,6 +118,9 @@ def handler(signum, frame):
 
 def is_container(obj: Any) -> bool:
     """Test if an object is a container (iterable) but not a string."""
+    if isinstance(obj, Proxy):
+        # unwrap before checking __iter__
```

Comment:
The comment isn’t really meaningful. Either it should explain _why_ unwrapping is needed, or should be removed entirely.

Reply:
I see. I wrote 2 candidates. Are either of these acceptable?
```suggestion
        # To prevent a Proxy from being determined as a container, unwrap obj before checking __iter__
```

```suggestion
```

Reply:
The first is better than the current one but not good enough; it should explain why the Proxy would be considered a container if not treated specially (because it implements `__iter__` to forward the call to the lazily initialised object). The second is obviously OK.

Reply:
```suggestion
        # Proxy of any object is considered a container because it implements __iter__ to forward the call to the lazily initialized object
        # Unwrap Proxy before checking __iter__ to evaluate the proxied object 
```